src/org/zaproxy/zap/extension/viewstate/ExtensionHttpPanelViewStateView.java

```diff
@@ -0,0 +1,273 @@
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.HttpPanelViewModelUtils;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.request.RequestBodyByteHttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.request.RequestByteHttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.request.RequestHeaderByteHttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.response.ResponseBodyByteHttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.response.ResponseByteHttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.response.ResponseHeaderByteHttpPanelViewModel;
+import org.zaproxy.zap.extension.viewstate.ViewStateModel.ViewStateUpdatedListener;
+import org.zaproxy.zap.extension.brk.BreakpointMessageHandler;
+import org.zaproxy.zap.extension.brk.ExtensionBreak;
+import org.zaproxy.zap.extension.httppanel.ComponentChangedEvent;
+import org.zaproxy.zap.extension.httppanel.MessagePanelEventListener;
+import org.zaproxy.zap.extension.httppanel.MessageViewSelectedEvent;
+import org.zaproxy.zap.extension.httppanel.component.all.request.RequestAllComponent;
+import org.zaproxy.zap.extension.httppanel.component.all.response.ResponseAllComponent;
+import org.zaproxy.zap.view.HttpPanelManager;
+import org.zaproxy.zap.view.HttpPanelManager.HttpPanelViewFactory;
+
+import net.htmlparser.jericho.Attribute;
+import net.htmlparser.jericho.Attributes;
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.HTMLElementName;
+import net.htmlparser.jericho.Source;
+
+public class ExtensionHttpPanelViewStateView extends ExtensionAdaptor {
+	
+	public static final int VS_ACTION_REQUEST = 1;
+	public static final int VS_ACTION_RESPONSE = 2;
+	private static final List<Class<?>> EXTENSION_DEPENDENCIES;
+	static {
+		// Prepare a list of Extensions on which this extension depends
+		List<Class<?>> dependencies = new ArrayList<>(1);
+		dependencies.add(ExtensionBreak.class);
+		dependencies.add(ExtensionHttpPanelHexView.class);
+		EXTENSION_DEPENDENCIES = Collections.unmodifiableList(dependencies);
+	}
+    private static Logger logger = Logger.getLogger(ExtensionHttpPanelViewStateView.class);
+	public static final String NAME = "ExtensionHttpPanelViewStateView";
+	public static final int PROXY_LISTENER_ORDER = 10;
+	private ExtensionBreak eBreak;
+	private boolean breakAvailableAndEnabled = false;
+    private BreakpointMessageHandler breakpointMessageHandler;
+    private HttpPanelViewStateView reqView;
+    private HttpPanelViewStateView respView;
+    private ViewStateModel reqModel;
+    private ViewStateModel respModel;
+	
+	public ExtensionHttpPanelViewStateView() {
+		super(NAME);
+		initialize();
+	}
+	
+	public ExtensionHttpPanelViewStateView(String name) {
+        super(name);
+        initialize();
+    }
+
+    private void initialize() {
+        this.setName(NAME);

```

Comment:
This statement can be removed (the name is already being set in the constructor).